### PR TITLE
unpin button improvement

### PR DESCRIPTION
**Pull request by RobertWarrenGilmore**
The unpin button for vertices is now an icon of a push pin and it only appears when the user hovers the mouse over the vertex.

### CODE REVIEW

File: lib/viewcomponents/unpinButton.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+var React = require('react');
+
+var pinImagePath = 'M19.5,37l10,0l1.07178,24.5l3.01953,0l0.908691,-24.5l10.5,0c0,0 -0.332361,-10.3673 -7.5,-10.5c0.272303,-18.12 0,-18.5 0,-18.5l2,0l0,-3.5l-15,0l0,3.5l2,0c0,0 -0.039881,0.0580001 0,18.5c-7.62955,0.0191484 -7,10.5 -7,10.5Z';
+var pinImageTransform = 'rotate(30) scale(0.25) translate(-32, -32)';
+
+var UnpinButton = React.createClass({
+
+  render: function() {
+    return (
+      <g className='unpinButton' {...this.props}>
```
Comment (jboolean) [inline lib/viewcomponents/unpinButton.jsx]: I think I'd been using hyphens-in-css-class names. We don't have a standard, maybe we should.

File: lib/viewcomponents/unpinButton.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+var React = require('react');
+
+var pinImagePath = 'M19.5,37l10,0l1.07178,24.5l3.01953,0l0.908691,-24.5l10.5,0c0,0 -0.332361,-10.3673 -7.5,-10.5c0.272303,-18.12 0,-18.5 0,-18.5l2,0l0,-3.5l-15,0l0,3.5l2,0c0,0 -0.039881,0.0580001 0,18.5c-7.62955,0.0191484 -7,10.5 -7,10.5Z';
```
Comment (jboolean) [inline lib/viewcomponents/unpinButton.jsx]: line too long. yikes.

File: lib/viewcomponents/vertex.jsx [hunk 3]
```diff
@@ -33,13 +34,22 @@ DIRECTION_TO_LABEL_POSITION[CardinalDirection.WEST] = PortLabelPosition.RIGHT;
 
 var Vertex = React.createClass({
 
+  getInitialState: function() {
+    return {
+      mouseHover: false
```
Comment (jboolean) [inline lib/viewcomponents/vertex.jsx]: This feels like the wrong place to be doing hovering stuff. 
Can the UnpinButton manage hovering itself? Better yet, can you just do this in CSS?

File: lib/viewcomponents/unpinButton.jsx (hunk 1 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/unpinButton.jsx]: Agreed. I'll observe our informal standard and change this.

File: lib/viewcomponents/unpinButton.jsx (hunk 2 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/unpinButton.jsx]: This string has to be easily replaceable, without having to add in a whole bunch of `'\n+ '`s It's a path description for a nontrivial shape.

Comment (RobertWarrenGilmore) [inline lib/viewcomponents/unpinButton.jsx]: I think that this is one case in which a long line is permissible.

File: lib/viewcomponents/vertex.jsx (hunk 3 above)
Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: Attempting in CSS.

Comment (RobertWarrenGilmore) [inline lib/viewcomponents/vertex.jsx]: Fixed. I was mistaken about this. There's a similar thing that CSS selectors can't do, but this isn't it.

File: examples/public/stylesheets/draw/vertex.less [hunk 4]
```diff
@@ -11,4 +11,8 @@
     color: black;
     alignment-baseline: central;
   }
+
+  &:not(:hover) > .unpin-button {
+    display: none;
+  }
```
Comment (RobertWarrenGilmore) [inline examples/public/stylesheets/draw/vertex.less]: @jboolean, better?

Comment (jboolean) [inline examples/public/stylesheets/draw/vertex.less]: Nice

**Comment by jboolean** [comment]
**Approve** for now. 
So far I don't think we've thought much about visual design. At some point later we may want to polish things up, but this general functionality will stay.

**Pull request merged**